src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/InductionScheduleUpdateEventPusher.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import uk.gov.justice.hmpps.sqs.HmppsTopic
+import uk.gov.justice.hmpps.sqs.publish
+import java.net.URI
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@Component
+class InductionScheduleUpdateEventPusher(
+  private val hmppsQueueService: HmppsQueueService,
+  private val objectMapper: ObjectMapper,
+  @Value("\${service.base-url}") private val serviceBaseUrl: String,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
```

Comment:
Minor, but we use `KotlinLogging` to get our logger instance, and we do it outside of the class definition:
```
package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging

import mu.KotlinLogging
import org.springframework.stereotype.Service
import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReleasedAdditionalInformation

private val log = KotlinLogging.logger {}

@Service
class PrisonerReleasedFromPrisonEventService {
```

Why? I hear you ask 🤣 
I believe its a) more kotlin specific, and the various log methods can take lambas etc
and b) (probably more important) is the fully qualified class name that the logs appear under.
With your logger in the companion object, I believe your log statements will be for:
```
uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.InductionScheduleUpdateEventPusher$Companion
```
IE. it has `$Companion` suffixed at the end because the JVM doesnt know what a companion object is and sees it as a nested class (I think)

( We use a slightly outdated version, where the gradle dependency is from `io.github.microutils` - the newer version is from `io.github.oshai`, but it's the same thing
https://github.com/oshai/kotlin-logging/wiki )

Reply:
yeah copy paste I'll sort it